### ISSUE TITLE
[FR] Add other controls

### ISSUE THREAD

**Issue by ildar** [42SK/TVKILL#1]
Natural improvement is to add more remote "buttons". The one I miss most is TV/AV switch.
Thanks!

**Comment by 42SK** [comment]
IR-patterns that do other things than turning the TV off are probably an important feature that should be added soon.
However, one should keep in mind that the main goal of **TVKILL** is not to become a fully featured IR-Remote with all the capabilities of a regular remote. **TVKILL** is intended to be a amusement tool that can either be used to play pranks on others (by turning off TVs) or to encourage responsible TV watching (**TVKILL** was mostly inspired by the [TV-B-Gone](https://wikipedia.org/wiki/TV-B-Gone)-Remote).
Any new "buttons" should therefore add a feature that might prove to be useful in this matter (e.g. a "mute" button).
Requests for new controls ("buttons") that would help to enhance the above-mentioned purpose can from now on be made in this thread.

**Comment by dan2468** [comment]
+1 for mute.
Maybe add a “learn” feature to record a new off pattern.

**Comment by 42SK** [comment]
@dan2468
A feature that lets you record patterns is probably not possible, since Android devices usually aren't equipped with an IR receiver.
We could add a feature that let's you enter IR codes manually, but users would have to get the patterns elsewhere, which wouldn't be very practical, since IR databases on the Internet are quite rare.
Adding a mute button should be less complicated though ;-) .

**Comment by ildar** [comment]
I think LIRC project's data may fit. It is open as FOSS. JFYI

пн, 16 нояб. 2015, 23:32, 42SK <notifications@github.com>:

> @dan2468 <https://github.com/dan2468>
> A feature that lets you record patterns is probably not possible, since
> Android devices usually aren't equipped with an IR receiver.
> We could add a feature that let's you enter IR codes manually, but users
> would have to get the patterns elsewhere, which wouldn't be very practical,
> since IR databases on the Internet are quite rare.
> Adding a mute button should be less complicated though ;-) .
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/42SK/TVKILL/issues/1#issuecomment-157111231>.
>
-- 
Ildar Mulyukov,
child of God

email: ildar.mulyukov@gmail.com
GoogleTalk: ildar.mulyukov@gmail.com
blog: http://johan-notes.blogspot.com/

**Comment by 42SK** [comment]
@ildar 
Thanks a lot for your suggestion! The LIRC database looks quite promising.

**Comment by a-raccoon** [comment]
Thanks a bunch! This is quite possibly the first (working) FOSS IR Blaster app out there, at least in my 3 years of digging.  Much appreciated!  I would love to see the project forked and expanded upon as suggested above, but this app is a gem as is.

As a thought, would it be possible to detect the impulses of a standard television remote from our device's camera, which CAN sense IR even through our len's cheap IR filter, or are the patterns too rapid to decode?

**Comment by 42SK** [comment]
@a-raccoon
Yes, the patterns are probably to rapid to be identified with a normal camera, but it seems to be possible to record the IR patterns of almost any remote with an IR receiver and an Arduino (this was suggested in issue #7). We might be able to use this in order to decode the patterns which could then be [added to **TV KILL**](https://github.com/42SK/TVKILL/wiki/How-to-add-IR-patterns-to-TV-KILL), assuming that we can find somebody who has experience with Arduino boards and IR codes.

**Comment by 42SK** [comment]
I've added a mute button in [version 1.1](https://github.com/42SK/TVKILL/releases/tag/v1.1) (0e5f28defc52665754928f8ba8c0bd92ade3fb46).
Have fun! :-)

**Comment by NeverMine17** [comment]
I think it's full remote not needed in app named TV **KILL**.
Although its might be useful, it will ruin the main idea of app.
